src/pcap/cmd/pcap-bosh-cli/main.go

```diff
@@ -0,0 +1,285 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"os"
+	"os/signal"
+	"regexp"
+
+	"github.com/cloudfoundry/pcap-release/src/pcap"
+	"github.com/cloudfoundry/pcap-release/src/pcap/bosh"
+
+	"github.com/jessevdk/go-flags"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	logger         *zap.Logger
+	atomicLogLevel zap.AtomicLevel
+)
+
+type options struct {
+	File               string   `short:"o" long:"file" description:"The output file. Written in binary pcap format." required:"true"`
+	PcapAPIURL         string   `short:"u" long:"pcap-api-url" description:"The URL of the PCAP API, e.g. pcap.cf.$LANDSCAPE_DOMAIN" env:"PCAP_API" required:"true"`
+	Filter             string   `short:"f" long:"filter" description:"Allows to provide a filter expression in pcap filter format." required:"false"`
+	Interface          string   `short:"i" long:"interface" description:"Specifies the network interface to listen on." default:"eth0" required:"false"`
+	Type               string   `short:"t" long:"type" description:"Specifies the type of process to capture for the app." default:"web" required:"false"`
+	BoshConfigFilename string   `short:"c" long:"bosh-config" description:"Path to the BOSH config file, used for the UAA Token" default:"${HOME}/.bosh/config" required:"false"`
+	BoshEnvironment    string   `short:"e" long:"bosh-environment" description:"The BOSH environment to use for retrieving the BOSH UAA token from the BOSH config file" default:"bosh" required:"false"`
+	Deployment         string   `short:"d" long:"deployment" description:"The name of the deployment in which you would like to capture." required:"true"`
+	InstanceGroups     []string `short:"g" long:"instance-group" description:"The name of an instance group in the deployment in which you would like to capture. Can be defined multiple times." required:"true"`
+	InstanceIds        []string `positional-arg-name:"ids" description:"The instance IDs of the deployment to capture." required:"false"`
+	Verbose            []bool   `short:"v" long:"verbose" description:"Show verbose debug information"`
+}
+
+// TODO: still too long but how can we split it into multiple methods?
+func main() {
+	var (
+		err  error
+		opts options
+	)
+	setupLogging()
+
+	defer func() {
+		if err != nil {
+			logger.Panic("execution failed", zap.Error(err))
+		}
+	}()
+
+	// Parse command-line arguments
+	_, err = flags.ParseArgs(&opts, os.Args[1:])
```

Comment:
Because we're not doing anything special with the args before parsing, we can also just use the default function `Parse()` here. It automatically takes `os.Args`.
```suggestion
	_, err = flags.Parse(&opts)
```